drivers/serial/uart_native_posix.c

```diff
@@ -261,9 +266,20 @@ static void np_uart_poll_out(const struct device *dev,
 				      unsigned char out_char)
 {
 	int ret;
-	struct native_uart_status *d;
+	struct native_uart_status *d = (struct native_uart_status *)dev->data;
+	struct pollfd pfd = { .fd = d->out_fd, .events = POLLHUP };
+
+	if (wait_pts) {
+		while (1) {
+			poll(&pfd, 1, 0);
+			if (!(pfd.revents & POLLHUP)) {
+				/* There is now a reader on the slave side */
+				break;
+			}
+			k_sleep(K_MSEC(100));
```

Comment:
Do you really want to stall only this thread, here?
Or do you want to halt the whole native_posix until something connects to the the slave side of the UART?

Reply:
Yes, only this thread.

Reply:
how come? meaning, what is the use case?

Reply:
Because if other threads really need to wait for this UART pts readiness, then they may be synchronized in application using semaphores or other IPC. Let's turn the question, what are benefits to halt whole native_posix (if I understand the question correctly)?

Reply:
native_posix is normally used as a test platform. Waiting until something is connected to the UART could be beneficial for example as part of an automated test. In that case, it would probably be desirable to have the whole test not be started until the other side is connected instead of having native_posix run ahead. It would make the test much more controllable and deterministic.
Therefore I ask, what use case do you have in mind in which you want to wait for another process, yet you think is better to have native_posix run for an unknown amount of time before the other end connects?

Reply:
Ok understand. I have a little bit different use case, where separated thread starts python interpreter and its output is routed to UART/pts independent on other threads already running and doing something.

Maybe for the case you mentioned should be some configuration option.

Reply:
If you have a specific use for this it can be this way. For the other use (holding the whole native_posix), if you want to add an option it would be nice. Otherwise somebody else can do it later when they have the need.

Reply:
I'd rather leave it on another PR.